lib/FtpContext.js

```diff
@@ -212,31 +252,43 @@ module.exports = class FTPContext {
      * will hold whatever the handler passed on when resolving/rejecting its task.
      *
      * @param {string} command
-     * @param {ResponseHandler} handler
+     * @param {ResponseHandler} responseHandler
      * @returns {Promise<any>}
      */
-    handle(command, handler) {
-        if (this._handler !== undefined) {
-            this._closeWithError(new Error("There is still a task running. Did you forget to use '.then()' or 'await'?"));
+    handle(command, responseHandler) {
+        if (this._task) {
+            // The user or client instance called `handle()` while a task is still running.
+            const err = new Error("User launched a task while another one is still running. Forgot to use 'await' or '.then()'?");
+            err.stack += `\nRunning task launched at: ${this._task.stack}`;
+            this._closeWithError(err);
         }
-        // Only track control socket timeout during the lifecycle of a task associated with a handler.
-        // That way we avoid timeouts on idle sockets, a behaviour that is not expected by most users.
-        this.enableControlTimeout(true);
         return new Promise((resolvePromise, rejectPromise) => {
-            this._handler = handler;
-            this._task = {
-                // When resolving or rejecting we also want the handler
-                // to no longer receive any responses or errors.
+            const resolver = {
                 resolve: (...args) => {
                     this._stopTrackingTask();
                     resolvePromise(...args);
                 },
-                reject: (...args) => {
+                reject: err => {
                     this._stopTrackingTask();
-                    rejectPromise(...args);
+                    rejectPromise(err);
                 }
             };
-            if (command !== undefined) {
+            this._task = {
+                stack: new Error().stack,
+                resolver,
+                responseHandler
+            };
+            if (this._closingError) {
+                // This client has been closed. Provide an error that describes this one as being caused
+                // by `_closingError`, include stack traces for both.
+                const err = new Error(`Client has been closed: ${this._closingError.message}`);
+                err.stack += `\nCaused by: ${this._closingError.stack}`;
```

Comment:
I was thinking that if someone just logs the error object using `toString` or just `message`, it might be helpful to include the cause there. Not sure though, I agree it would be cleaner the other way.

Reply:
My vote is to keep message + stack together.  That's how other "long" stacks do it, so less confusing.